### PR TITLE
Output actual program + more efficient Hamming weight

### PR DESCRIPTION
**Pull request by Ko-**
Hi. Not sure if you're aiming at a particular minimal Python version to support or at a particular coding style. Otherwise I could still fix that.

### CODE REVIEW

File: heuristic.py [hunk 1]
```diff
@@ -103,7 +107,7 @@ def D(S, i=None):
 
 def D_(S, i):
     if not current_weights:
-        mindist = sum([1 for a in bin(M[i]) if a == '1'])
+        mindist = bin(M[i]).count('1')
```
Comment (thomwiggers) [inline heuristic.py:110]: :+1:

File: heuristic.py [hunk 2]
```diff
@@ -177,24 +182,28 @@ def find_next_base(minsum):
             options.append((newrow, weights))
 
     if len(options) == 1:
-        return options[0][0]
+        program.append(('t{}'.format(len(program)), '{} ^ {}'.format(program[options[0][0][1]][0], program[options[0][0][2]][0])))
```
Comment (thomwiggers) [inline heuristic.py]: I'd prefer to stick to pep8, so a maximum line length of 80 chars.

**Review by thomwiggers — CHANGES_REQUESTED** [review]
Thanks a lot, seems like a good improvement. Would be great if you could run it through `pep8` or `flake8`.

**Comment by Ko-** [comment]
I ran stuff through autopep8.

**Review by thomwiggers — APPROVED** [review]

**Pull request merged**